### PR TITLE
Fix for --unauthenticated-client-action V2 issue

### PR DESCRIPTION
**Pull request by annzho**
Previously, using `az webapp auth --unauthenticated-client-action` argument did not correctly update Unauthenticated Client Action configuration setting **if the argument was set to RejectWith401, RejectWith404, or RejectWith403** due a naming mismatch between the CLI and the API surface. 

Previous issues raised for this: https://github.com/Azure/azure-cli/issues/24086, https://github.com/Azure/azure-cli/issues/21618

The fix was to rename some of the allowed values for --unauthenticated-client-action. Also added the Return403 value. Allowed Values: RedirectToLoginPage, AllowAnonymous, Return401, Return404, Return403

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command
`az webapp auth update` command fix --unauthenticated-client-action argument

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repository and upgrade the version in the pull request but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
webapp

**Comment by annzho** [comment]
Rerecorded the live tests, CI is now passing. @zhoxing-ms, could you help with approving?

**Comment by zhoxing-ms** [comment]
@annzho Please add the description of PR into the `HISTORY.rst`. If you want to release a new version for this PR, please also upgrade the version defined in the `setup.py`.

**Comment by annzho** [comment]
@zhoxing-ms Have added the description of the new version in `HISTORY.rst` and upgraded version in `setup.py.` Thanks!

File: src/authV2/HISTORY.rst [hunk 1]
```diff
@@ -5,4 +5,8 @@ Release History
 
 0.1.0
 ++++++
-* Initial release.
\ No newline at end of file
+* Initial release.
+
+0.1.2
+++++++
+* Bug fix for '--unauthenticated-client-action' option in webapp auth, updated allowed values.
```
Comment (zhoxing-ms) [inline src/authV2/HISTORY.rst:12]: Please add the new history notes to the top of the file

Comment (annzho) [reply src/authV2/HISTORY.rst]: Just moved it!

**Comment by annzho** [comment]
@zhoxing-ms, moved the version notes to top of history file, could you approve?

**Pull request merged**